### PR TITLE
fix: remove resourceowner read from context in user v2 api

### PR DESCRIPTION
**Pull request by stebenz**
Remove logic from user V2 API to make calls resource based, accidentally the organization of the user was still considered in some calls, so that the only necessary information is the user ID.

Relates #7253

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [x] All open todos and follow ups are defined in a new ticket and justified (https://github.com/zitadel/zitadel/issues/7253#issuecomment-2023392302)
- [x] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [x] Critical parts are tested automatically
- [x] Where possible E2E tests are implemented
- [x] Documentation/examples are up-to-date
- [x] All non-functional requirements are met
- [x] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: internal/command/user_idp_link.go [hunk 1]
```diff
@@ -15,15 +15,22 @@ func (c *Commands) AddUserIDPLink(ctx context.Context, userID, resourceOwner str
...
+		return nil, zerrors.ThrowPreconditionFailed(nil, "COMMAND-vzktar7b7f", "Errors.User.NotFound")
+	}
 	if userID != authz.GetCtxData(ctx).UserID {
-		if err := c.checkPermission(ctx, domain.PermissionUserWrite, resourceOwner, userID); err != nil {
+		if err := c.checkPermission(ctx, domain.PermissionUserWrite, existingUser.ResourceOwner, existingUser.AggregateID); err != nil {
 			return nil, err
 		}
 	}
-	event, err := addLink(ctx, c.eventstore.Filter, user.NewAggregate(userID, resourceOwner), link)
+	// We ignore the deprecation warning here because we still need to support the deprecated field.
```
Comment (livio-a) [inline internal/command/user_idp_link.go:31]: ignoring the deprecation warning for the filter is ok at the moment, but what do you mean by `support the deprecated field`?

File: internal/command/user_idp_link.go [hunk 2]
```diff
@@ -15,15 +15,22 @@ func (c *Commands) AddUserIDPLink(ctx context.Context, userID, resourceOwner str
 	if userID == "" {
 		return nil, zerrors.ThrowInvalidArgument(nil, "COMMAND-03j8f", "Errors.IDMissing")
 	}
-	if err := c.checkUserExists(ctx, userID, resourceOwner); err != nil {
+
```
Comment (livio-a) [inline internal/command/user_idp_link.go:18]: among others, the function is called from the user v2 service and that still passes the org id from context

Comment (livio-a) [reply internal/command/user_idp_link.go:18]: and i noticed the `authz.GetCtxData(ctx).OrgID` is also still used elsewhere in that service, is that intended?

**Review by livio-a — CHANGES_REQUESTED** [review]
please provide a description suitable for clients and has this never been tested manually?
and i think it relates / should close [#7253](https://github.com/zitadel/zitadel/issues/7253)?

File: internal/command/user_idp_link.go (hunk 2 above)
Comment (stebenz) [reply internal/command/user_idp_link.go:18]: No, that's exactly the spot where I wanted to remove it from, is removed.

File: internal/command/user_idp_link.go (hunk 1 above)
Comment (livio-a) [reply internal/command/user_idp_link.go:31]: i now search for that and the only thing i could find was a part where we specifically need access to a deprecated `GetCreationDate()`, which was deprecated in the API and has nothing to do with the deprecation of the filter

**Review by livio-a — CHANGES_REQUESTED** [review]
as mentioned before: please provide a description suitable for clients
e.g. explain that the v2 api is resource based but accidentially the organization was still considered on some request, ...

**Comment by stebenz** [comment]
@livio-a description changed an comment removed

**Review by livio-a — APPROVED** [review]

**Pull request merged**